src/main/java/org/assertj/core/api/RecursiveIterableComparisonAssert.java

```diff
@@ -0,0 +1,285 @@
+   * By default, {@link #isEqualTo(Iterable) isEqualTo} takes element order into consideration.
+   * This means that a List and a Set can not be considered equals, even if they contain exactly the same elements.
+   * To ignore the element order when comparing iterables, call {@link #ignoringActualIterableOrder() ignoringActualIterableOrder}.
+   *
+   * <pre><code class='java'>public class Person {
+   *    String name;
+   *    boolean hasPhd;
+   * }
+   *
+   * public class Doctor {
+   *    String name;
+   *    boolean hasPhd;
+   * }
+   *
+   * Person sheldon = new People("Sheldon", true);
+   * Person leonard = new People("Leonard", true);
+   * Person howard = new People("Howard", false);
+   *
+   * Doctor drSheldon = new Doctor("Sheldon", false);
+   * Doctor drLeonard = new Doctor("Leonard", false);
+   *
+   *
+   * List<Person> peopleList = Arrays.asList(sheldon, leonard, howard);
+   * Set<Person> peopleLinkedSet = new LinkedHashSet<>(peopleList);
+   *
+   * // assertion succeeds because both iterables have same items in same order.
+   * assertThat(peopleList).usingRecursiveComparison()
+   *                       .isEqualTo(peopleLinkedSet);
+   *
+   *
+   * Set<Person> peopleHashSet = new HashSet<>(peopleList);
+   *
+   * // assertion fails even though iterables have same items, the order is different.
+   * assertThat(peopleList).usingRecursiveComparison()
+   *                       .allowingArrayTypeForExpected()
+   *                       .isEqualTo(peopleHashSet);
+   *
+   * // assertion succeeds because the order is ignored.
+   * assertThat(peopleList).usingRecursiveComparison()
+   *                       .ignoringActualIterableOrder()
+   *                       .isEqualTo(peopleHashSet);
+   *
+   * // assertion fails because iterables are restricted to be compatible by {@link #withStrictTypeCheckingOnActualIterable() withStrictTypeCheckingOnActualIterable}.
+   * assertThat(peopleList).usingRecursiveComparison()
+   *                       .withStrictTypeCheckingOnActualIterable()
+   *                       .isEqualTo(peopleLinkedSet);</code></pre>
+   *
+   * @param expected the object to compare {@code actual} to.
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual and the given objects are not deeply equal property/field by property/field.
+   * @throws IntrospectionError if one property/field to compare can not be found.
+   */
+  @SuppressWarnings("unchecked")
+  public SELF isEqualTo(Iterable<?> expected) {
+    if((actual == null && expected == null) || (actual == expected))
+        return myself;
+
+    if(actual == null || expected == null)
+      throw new AssertionError("Iterables to compare can not be null.");
+
+    if(recursiveComparisonConfiguration.isInStrictTypeCheckingOnActualMode() &&  !actual.getClass().isAssignableFrom(expected.getClass()))
+      throw new AssertionError("types are not compatible.");
```

Comment:
The error message must more informative and show the types, a bit like the one in `RecursiveComparisonAssert`:
```
"the fields are considered different since the comparison enforces strict type check and %s is not a subtype of %s"
```